sdk/cosmos/azure-cosmos/azure/cosmos/auto_scale.py

```diff
@@ -0,0 +1,33 @@
+# The MIT License (MIT)
+# Copyright (c) 2022 Microsoft Corporation
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+class AutoScaleProperties(object):
+    """Represents the auto-scale throughput properties in an Azure Cosmos DB SQL API container.
+
+   :param auto_scale_max_throughput is the max autoscale throughput, it should have valid throughput
+   values between 1000 and 1000000 inclusive in increments of 1000
+
+   :param auto_upgrade_throughput_increment_percent is the auto upgrade max throughput increment percentage
+   default value is 0 the increment percent should be greater than or equal to zero
+    """
+    def __init__(self, auto_scale_max_throughput, auto_upgrade_throughput_increment_percent=0):
+        self.auto_scale_max_throughput = auto_scale_max_throughput
+        self.auto_upgrade_throughput_increment_percent = auto_upgrade_throughput_increment_percent
```

Comment:
These aren't actual header names - any chance we could improve the names?
They don't need the word "auto" in them - that's already implied by the object name.
These names are long - yet their length doesn't provide me much clarity as to what I'm configuring...so I'm reliant on documentation anyway...

Reply:
Another thought - do we expect the number of properties here to increase?
I'm not _that_ opposed to using a model for this (rather than flattening it out) - but only 2 properties is a pretty fine line....
How do these two properties interact? Must I always set both of them? It looks like the 2nd one has a default value - will most people use this? If so, are we making most people instantiate an object to set a single property?

Reply:
I am beginning to feel that we should just put the properties in our Offer file that contains the ThroughputProperties but I am not sure.